packages/core/src/runtime/utils/hashPropertyKey.ts

```diff
@@ -3,9 +3,22 @@ import hash from '@emotion/hash';
 import type { PropertyHash } from '../../types';
 import type { AtRules } from './types';
 
+function addAtRulePrefix(atRule: string, prefix: string): string {
+  return atRule ? prefix + atRule : atRule;
+}
+
+export function atRulesToString(atRules: AtRules): string {
+  return (
+    addAtRulePrefix(atRules.container, 'c') +
+    addAtRulePrefix(atRules.media, 'l') +
+    addAtRulePrefix(atRules.layer, 'l') +
+    addAtRulePrefix(atRules.supports, 's')
+  );
+}
+
 export function hashPropertyKey(selector: string, property: string, atRules: AtRules): PropertyHash {
   // uniq key based on property & selector, used for merging later
-  const computedKey = selector + atRules.container + atRules.media + atRules.supports + property;
+  const computedKey = selector + atRulesToString(atRules) + property;
```

Comment:
previously layer queries were not supported in `hashPropertyKey` I guess it was unintentional?

Reply:
No, it's a bug https://github.com/microsoft/griffel/issues/561.